### PR TITLE
Sort feature

### PR DESCRIPTION
**Pull request by Nathan-O**
Adds sort feature to the api_me library. Now by passing in certain parameters the model can be sorted before it's given to the route or paginated.

### CODE REVIEW

File: lib/api_me/pagination.rb [hunk 1]
```diff
@@ -30,15 +29,15 @@ def page_meta
     protected
 
     def page
-      self.scope = scope.page(self.page_offset ? page_offset : 1)
+      self.scope = Kaminari.paginate_array(scope).page(self.page_offset ? page_offset : 1)
+      # self.scope = scope.page(self.page_offset ? page_offset : 1)
```
Comment (chadzink) [inline lib/api_me/pagination.rb]: take this out before merge

File: lib/api_me/sorting.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        reverse: sort_reverse,
+        record_count: scope.size,
+        total_records: scope.total_count,
+      }
+    end
+
+    protected
+
+    def sort
+      unless sort_criteria === ""
```
Comment (chadzink) [inline lib/api_me/sorting.rb]: maybe use `sort_criteria.is_blank?` instead?

**Comment by Samsinite** [comment]
Looks like you have a test failure on travis, needs fixed before merging.

File: spec/acceptance/api/v1/posts_spec.rb [hunk 3]
```diff
@@ -137,4 +137,30 @@
...
+    end
+
+    get '/api/v1/posts?sort%5Bcriteria%5D=id&sort%5Breverse%5D=true'
+    json = JSON.parse(last_response.body)
+    expect(json['posts'].last['name']).to eq('Post1')
+  end
+
+  it 'sends posts reverse sorted by id and paginated with a size of 12 and an offset of 1' do
+    i = 0
+    20.times do
```
Comment (Samsinite) [inline spec/acceptance/api/v1/posts_spec.rb]: You can do `20.times do |i|` here instead of `i=0` and `i += 1`

**Comment by Samsinite** [comment]
After fixing the failing test and addressing Chad's comment, :+1:. Looks good, nice job!

**Comment by chadzink** [comment]
Very nice work, :+1: after comments and test fixes

**Comment by Nathan-O** [comment]
@chadzink @Samsinite there you go, should be all good now.

**Comment by chadzink** [comment]
Good to merge imo

**Pull request merged**